### PR TITLE
docs: adds readme to tracking explaining how to test components with …

### PR DESCRIPTION
**Pull request by nikhedonia**

### CODE REVIEW

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/105381bf211cb864a1a1d78f29dde165727a2fd3/dextrose-test/index.html

File: packages/tracking/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Tracking
+
+Tracking is a collection of Higher-Order-Components (HOCs) that collects and
```
Comment (craigbilner) [inline packages/tracking/README.md]: this is quite hard to parse, perhaps an ASCII diagram?

File: packages/tracking/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+reports user defined events. All HOCs consumes a component and an object
```
Comment (craigbilner) [inline packages/tracking/README.md]: typo: consume

File: packages/tracking/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+describing how to create the analytics event.
+
+The returned Component expects an analyticsStream to be available in context or
```
Comment (craigbilner) [inline packages/tracking/README.md]: would use backticks for code emphasis

File: packages/tracking/README.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+it's analyticsStream to it's descendants via context.
+
+## Testing Components with Tracking
+
+The analyticsStream prop is expected to be a function that consumes an event
+object.
+
+To test whether your component propagates the correct events, pass `jest.fn()`
+as the `analyticsStream` and test if your mock received the expected events.
+Mind that you need to mock the `Date` and `setTimeout` to make your tests
```
Comment (craigbilner) [inline packages/tracking/README.md]: `setTimeout` is already taken care of by JC?

File: packages/tracking/README.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+The returned Component expects an analyticsStream to be available in context or
+to be passed directly to the component via props. Every tracking HOC passses
+it's analyticsStream to it's descendants via context.
+
+## Testing Components with Tracking
+
+The analyticsStream prop is expected to be a function that consumes an event
+object.
+
+To test whether your component propagates the correct events, pass `jest.fn()`
```
Comment (craigbilner) [inline packages/tracking/README.md]: perhaps point to a test example?

File: packages/tracking/README.md (hunk 4 above)
Comment (nikhedonia) [reply packages/tracking/README.md]: not yet

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/3686ff5ef86ee4f6b4a2b58563c5b91aae500a09/dextrose-test/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/b4d7b1345cb571bdd92e0b45a13ca2b1aaa30070/dextrose-test/index.html
